sql/parse/dateparse/date.go

```diff
@@ -28,17 +36,30 @@ func ParseDateWithFormat(date, format string) (time.Time, error) {
 	date = strings.ToLower(date)
 
 	var result datetime
+	var types = map[ParseType]bool{Time: false, Date: false, None: false}
 	target := date
 	for _, parser := range parsers {
 		target = takeAllSpaces(target)
-		rest, err := parser(&result, target)
+		rest, isTime, err := parser(&result, target)
```

Comment:
(minor) `isTime` makes me think of a boolean. `type` or `datetimeType` or something similar might be a little more descriptive of a name.